spec/determinator/retrieve/http_retreiver_spec.rb

```diff
@@ -0,0 +1,90 @@
+require 'determinator/retrieve/http_retriever'
+require 'spec_helper'
+require 'webmock/rspec'
+
+RSpec.describe Determinator::Retrieve::HttpRetriever do
+  describe '#retrieve' do
+    subject(:retrieve) { described_class.new(params).retrieve(feature_id) }
+    let(:client){
+      Faraday.new(base_url)
+    }
+    let(:base_url) { 'http://actortracking.dev' }
+    let(:service_name) { 'MY-SERVICE' }
+    let(:feature_id) { 'some-feature' }
+    let(:feature_json) { {
+      name: "Feature one",
+      identifier: "feature",
+      bucket_type: "id",
+      target_groups: [{
+        rollout: 65536,
+        constraints: {}
+      }],
+      active: true,
+      overrides: {}
+    } }
+    let(:expected_url) { "#{base_url}/features/#{feature_id}" }
+
+    shared_examples 'retrieve tests' do
+      before do
+        allow(Determinator).to receive(:notice_error)
+        stub_request(:get, expected_url).
+          to_return(status: response_status, body: feature_json.to_json)
+      end
+
+      context 'when the feature is found' do
+        let(:response_status) { 200 }
+
+        context 'when there is no other error' do
+          it 'returns a Feature object' do
+            expect(retrieve).to be_a(Determinator::Feature)
+          end
+
+          it 'sets the properties on the object' do
+            expect(retrieve.name).to eql(feature_json[:name])
+          end
+        end
+
+        context 'when an error occurs' do
+          let(:error) { StandardError.new }
+
+          before do
+            allow(Determinator::Serializers::JSON).to receive(:load).and_raise(error)
+            retrieve
+          end
+
+          it 'logs the error' do
+            expect(Determinator).to have_received(:notice_error).with(error)
+          end
+
+          it { is_expected.to be nil }
+        end
+      end
+
+      context 'when the feature is not found' do
+        let(:response_status) { 404 }
+
+        before do
+          retrieve
+        end
+
+        it { is_expected.to be nil }
+      end
+    end
+
+    context 'when client is injected' do
+      let(:params) { { connection: client } }
+
+      context 'when the client is a Faraday connection' do
+        include_examples 'retrieve tests'
```

Comment:
Suggestion: inline shared examples

I mean, I don't know if there is a plan to reuse these examples somewhere else, but I can only see one usage here, and they can add a bit of overhead when trying to understand test fails. So just a suggestion, I don't mind what you decide to do.

Reply:
I've split them out into a separate file, they're used by the dynaconf spec now too.